modules/apps/dynamic-data-mapping/dynamic-data-mapping-web/src/main/resources/META-INF/resources/js/ddm_form.js

```diff
@@ -2448,6 +2448,13 @@ AUI.add(
 							layoutPath.groupId,
 							layoutPath.privateLayout
 						);
+						if (index < layoutsPathLength - 1) {
+							var breadcrumbNode = instance._modal.bodyNode.one(
+								'.lfr-ddm-breadcrumb'
+							);
+
+							breadcrumbNode.append('&nbsp;&gt;&nbsp;');
```

Comment:
@javalosjr96, apparently, there is a chance of the `_modal` be `undefined`, as we can see [here](https://github.com/liferay-forms/liferay-portal/blob/90615e6f2359cd704ec938a87b52e5d7a0f38838/modules/apps/dynamic-data-mapping/dynamic-data-mapping-web/src/main/resources/META-INF/resources/js/ddm_form.js#L2381-L2383), thereby can you please make sure that the call `instance._modal.bodyNode.one` won't return us an `undefined` result? Otherwise, the call `breadcrumbNode.append` may fail.

Does it make sense to you?

Reply:
@natocesarrego 
I understand the issue, but besides a null check, how can I make it so it doesn't return an undefined result?

Reply:
Would following the same logic as `instance._addBreadcrumbElement(` solve this issue?

To Clarify, create a method called addBreadcrumbSpacer in the same instance as addBreadcrumbElement?

Reply:
@natocesarrego 
I've made changes that I think resolve the issue.  I basically followed the same logic as addBreadcrumbElement since it only exists if the instance is active.

Reply:
My point is how can we adjust the code to prevent the call `breadcrumbNode.append('&nbsp;&gt;&nbsp;');` fail?

I was thinking in something like:
```
var breadcrumbNode = instance._modal?.bodyNode?.one(
'.lfr-ddm-breadcrumb'
);

breadcrumbNode?.append('&nbsp;&gt;&nbsp;');
```
or
```
if (instance._modal && instance._modal.bodyNode) {
    var breadcrumbNode = instance._modal.bodyNode.one(
    '.lfr-ddm-breadcrumb'
    );

    if (breadcrumbNode) {
        breadcrumbNode.append('&nbsp;&gt;&nbsp;');
    }
}
```
The first suggestion won't work on 7.2 and prior versions since they have some restrictions regarding JavaScripts features, like the `?.` operator.

Reply:
@natocesarrego What I don't understand is why does it matter if the` instance._modal.bodyNode.one` returns undefined.  
If the the instances is unavailable, then nothing would work. What is the difference between calling `instance._modal.bodyNode.one` in `addBreadcrumbElement` and `addBreadcrumbSpacer`.  (using the latest commit as an example)

Reply:
Hi @javalosjr96, ideally, the same checks done by you should be done in `addBreadcrumbElement`. My biggest concern is we try to invoke a method / an attribute of an undefined object which will raise an uncaught exception, preventing the code move on with the execution as expected, is it clearer to you?

Sorry if I wasn't able to make it clearer before. :slightly_smiling_face: